### ISSUE TITLE
Improve HTML title of episode page

### ISSUE THREAD

**Issue by neingeist** [ephes/python-podcast#25]
Currently if I bookmark e.g. https://python-podcast.de/show/kubernetes/ I get a "Kubernetes" bookmark by default. I think it would be better if the HTML `<title>` said "Kubernetes (Python Podcast)" instead. Could also be a little better for searching in history (but there "python podcast" matches the URL anyway.)

(Related question: The `/show/` in the URL, does it refer to the episode? Then I would consider it wrong, the "show" is the "Python Podcast", while the episode is "Kubernetes". Naming things is hard, ask 2 devs, get 3 opinions.)

Please tell me if this kind of tiny nit pick is too nit-picky - I wasn't sure if I should report it but then decided you can still choose to "won't fix".

**Comment by ephes** [comment]
## First Part - Handling the html `title`-Element

Full ack for the html title (which is used as the name for a bookmark). Now for the html title the `page.seo_title` (promotion tab) attribute is used instead of `page.title` if available. For the [python-podcast](https://python-podcast.de/show) site ` (python-podcast)` is appended to all html titles. Thanks for suggesting this 😄.

**Comment by ephes** [comment]
## Second Part URL-Structure

The `/show/` path is not referring to the episode but to the page of the podcast. The `/show/kubernetes/` path points to the episode. It might be confusing because there's a redirect from `/` to `/show/` because this would come in handy if there are other podcasts added later. Then `/` could show a list of those podcasts. But since there is only one for now, `/` redirects to `show`.

Showing a list of episodes on the main podcast page is also debatable. A lot of podcasts just show the current episode and have a link to the archive somewhere. And showing a player for each episode is especially bad because it loads > 100kb javascript for each episode. Maybe there's a way to only load the player javascript only once, dunno.

**Comment by neingeist** [comment]
> Showing a list of episodes on the main podcast page is also debatable. A lot of podcasts just show the current episode and have a link to the archive somewhere. And showing a player for each episode is especially bad because it loads > 100kb javascript for each episode. Maybe there's a way to only load the player javascript only once, dunno.

From a listener perspective, it could be:

- have newest episode on the front page
- but the 4 next-newest als a list below that, including a "→ Older episodes" link 

Just don't hide "older episodes" link with a tiny font, I hate it when sites do that :smile:

**Comment by neingeist** [comment]
(Motivation: I listen to episodes in "random" order, and usually go to the podcast's site when I want show notes → should be easy to navigate to the right one. Unfortunately, my podcatcher is terrible in one thing: I don't get the episode URL easily)

**Comment by ephes** [comment]
> From a listener perspective, it could be:
> 
> * have newest episode on the front page
> * but the 4 next-newest als a list below that, including a "→ Older episodes" link

As a list of links / teasers but without player, right?
 
Hmm, sounds good. I already had a redesign in mind, this could be an idea..

**Comment by ephes** [comment]
> (Motivation: I listen to episodes in "random" order, and usually go to the podcast's site when I want show notes → should be easy to navigate to the right one. Unfortunately, my podcatcher is terrible in one thing: I don't get the episode URL easily)

Have you ever clicked on the "Filters/Fulltext Search" link to find an episode? It's probably a little to well hidden 😅.  I have the same "how to find the episode detail page"-problem. My podcatcher is [castro](https://castro.fm/) and its nearly impossible to find the episode url. And for some reason it misses new episodes lately. Maybe I have to switch 😥.

**Comment by neingeist** [comment]
> > From a listener perspective, it could be:
> > 
> > * have newest episode on the front page
> > * but the 4 next-newest als a list below that, including a "→ Older episodes" link
> 
> As a list of links / teasers but without player, right?
> 
> Hmm, sounds good. I already had a redesign in mind, this could be an idea..

Yeah something like that. Hard to say what would be great without trying it and not even drawing it on a napkin ^^

**Comment by neingeist** [comment]
> > (Motivation: I listen to episodes in "random" order, and usually go to the podcast's site when I want show notes → should be easy to navigate to the right one. Unfortunately, my podcatcher is terrible in one thing: I don't get the episode URL easily)
> 
> Have you ever clicked on the "Filters/Fulltext Search" link to find an episode? It's probably a little to well hidden 😅.

I'm not sure :) But it's definitely something I would use. With the Python podcast I have some unlistened episodes - ironically the ones that would be most relevant - and when I will listen to them I'm probably going to use the search to look up the episodes to get to the show notes. (I never ever use the show notes/details tab in my podcast app, never for any podcast, not on that tiny screen etc.)

> I have the same "how to find the episode detail page"-problem. My podcatcher is [castro](https://castro.fm/) and its nearly impossible to find the episode url. And for some reason it misses new episodes lately. Maybe I have to switch 😥.

Mine is Pocket Casts, the only issue I have with it is this :-) If you "share episode" the app uses their own pcasts.fm website - that's what I want in 0% of all cases, in 100% of all cases I would want to share a link to the episode page of the original site. Should probably open a bug there.

(The feed contains the actual episode URL in the right chunk of XML, I checked so I can blame the app guilt-free 😇)
